core/src/test/java/org/apache/calcite/sql/test/SqlTests.java

```diff
@@ -350,6 +350,16 @@ public static void checkEx(Throwable ex,
     }
     Throwable actualException = ex;
     String actualMessage = actualException.getMessage();
+
+    // If sap does not contain an error location, just check whether an exception
+    // matches the expected pattern.
+    if (sap.pos == null) {
+      if (expectedMsgPattern != null
+          && actualMessage.matches(expectedMsgPattern)) {
+        return;
```

Comment:
These conditions can be simplified to one if as follows.
```
if (sap.pos == null && expectedMsgPattern != null 
     && actualMessage.matches(expectedMsgPattern))
```

Reply:
changed